### PR TITLE
Support `NODE_TLS_REJECT_UNAUTHORIZED=0` at runtime and implement `BUN_CONFIG_VERBOSE_FETCH`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

- Fixes a bug with HTTPS keep-alive when `NODE_TLS_REJECT_UNAUTHORIZED=0` is set.
- Implements support for setting `process.env.NODE_TLS_REJECT_UNAUTHORIZED=0` at runtime
- Implements `BUN_CONFIG_VERBOSE_FETCH=1`

image

### How did you verify your code works?

There is a test for `NODE_TLS_REJECT_UNAUTHORIZED=0` and `NODE_TLS_REJECT_UNAUTHORIZED=1` and the keep-alive fix.

### CODE REVIEW

File: src/bun.js/bindings/JSEnvironmentVariableMap.cpp [hunk 1]
```diff
@@ -121,6 +122,135 @@ JSC_DEFINE_CUSTOM_SETTER(jsTimeZoneEnvironmentVariableSetter, (JSGlobalObject *
...
+JSC_DEFINE_CUSTOM_SETTER(jsNodeTLSRejectUnauthorizedSetter, (JSGlobalObject * globalObject, JSC::EncodedJSValue thisValue, JSC::EncodedJSValue value, PropertyName propertyName))
+{
+    VM& vm = globalObject->vm();
+    JSC::JSObject* object = JSValue::decode(thisValue).getObject();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    if (!object)
+        return false;
+
+    JSValue decodedValue = JSValue::decode(value);
+    if (decodedValue.isString()) {
```
Comment (cirospaciari) [inline src/bun.js/bindings/JSEnvironmentVariableMap.cpp]: how does node handle if the user pass like `process.env.NODE_TLS_REJECT_UNAUTHORIZED=0` (boolean or number) we probably should convert/coerce that to string here

File: src/bun.js/bindings/JSEnvironmentVariableMap.cpp [hunk 2]
```diff
@@ -121,6 +122,135 @@ JSC_DEFINE_CUSTOM_SETTER(jsTimeZoneEnvironmentVariableSetter, (JSGlobalObject *
...
+JSC_DEFINE_CUSTOM_SETTER(jsBunConfigVerboseFetchSetter, (JSGlobalObject * globalObject, JSC::EncodedJSValue thisValue, JSC::EncodedJSValue value, PropertyName propertyName))
+{
+    VM& vm = globalObject->vm();
+    JSC::JSObject* object = JSValue::decode(thisValue).getObject();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    if (!object)
+        return false;
+
+    JSValue decodedValue = JSValue::decode(value);
+    if (decodedValue.isString()) {
```
Comment (cirospaciari) [inline src/bun.js/bindings/JSEnvironmentVariableMap.cpp]: same here should we check for `number` and `boolean` too?

File: test/js/bun/http/bun-server.test.ts [hunk 3]
```diff
@@ -407,6 +407,25 @@ describe("Server", () => {
       const result = await fetch(`https://${url}`, { tls: { rejectUnauthorized: false } }).then(res => res.text());
       expect(result).toBe("Hello");
     }
+
+    // Test that HTTPS keep-alive doesn't cause it to re-use the connection on
+    // the next attempt, when the next attempt has reject unauthorized enabled
+    {
+      expect(
+        async () => await fetch(`https://${url}`, { tls: { rejectUnauthorized: true } }).then(res => res.text()),
```
Comment (cirospaciari) [inline test/js/bun/http/bun-server.test.ts]:
```suggestion
        async () => await fetch(server.url, { tls: { rejectUnauthorized: true } }).then(res => res.text()),
```

Comment (cirospaciari) [reply test/js/bun/http/bun-server.test.ts]: we probably should refactor to `const url = server.url.href;` that already includes the protocol (the same for the other places in this test)

**Review by cirospaciari — CHANGES_REQUESTED** [review]
few comments also missing tests for `WebSocket` client and `node:http` but if we already have tests for this and its working the fetch test for the runtime should be enough

File: src/bun.js/bindings/JSEnvironmentVariableMap.cpp (hunk 1 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/JSEnvironmentVariableMap.cpp]: https://github.com/nodejs/node/blob/78a326e2eb236874c45a7937d51e5d3ab3c91a3e/lib/internal/options.js#L50

It looks like they stringify and then check if the result is "0"

**Review by cirospaciari — APPROVED** [review]

**Pull request merged**